mypy/checkexpr.py

```diff
@@ -1543,8 +1544,11 @@ def visit_cast_expr(self, expr: CastExpr) -> Type:
         """Type check a cast expression."""
         source_type = self.accept(expr.expr, type_context=AnyType(), allow_none_return=True)
         target_type = expr.type
-        if self.chk.options.warn_redundant_casts and is_same_type(source_type, target_type):
+        options = self.chk.options
+        if options.warn_redundant_casts and is_same_type(source_type, target_type):
             self.msg.redundant_cast(target_type, expr)
+        if options.disallow_implicit_any_types and has_any_from_silent_import(target_type):
+            self.msg.implicit_any_from_silent_import("Target type of cast", target_type, expr)
```

Comment:
I'm not totally sure what you mean. I tried the following code and it gave an error (like it's supposed to).
```
from typing import cast, Callable
from missing import Unchecked

foo = [1, 2, 3]
call = cast(Callable[[], Unchecked], foo)
```
Produces the following output:
```
m.py:6: error: Target type of cast is implicitly converted to Callable[[], Any] due to import from unanalyzed module
```